### PR TITLE
async rendering doesn't throw anymore when encountering compilation errors

### PR DESCRIPTION
**Pull request by eventEmitter**
fixes http://github.com/mozilla/nunjucks/issues/233

when calling the `render` or `renderString` method with a callback and the template contains a syntax error an error is thrown instead of returning the error to the callback. This patch fixes this behavior. Compilation errors are now returned to the callback.

### CODE REVIEW

**Comment by garygreen** [comment]
Thanks for this. Merged :-)

File: src/environment.js [hunk 1]
```diff
@@ -342,7 +342,14 @@ var Template = Obj.extend({
         }
 
         return lib.withPrettyErrors(this.path, this.env.dev, function() {
-            this.compile();
+
+            // ctach compile errors for async rendering
```
Comment (popomore) [inline src/environment.js]: typo

Comment (garygreen) [inline src/environment.js]: Yup, spelling in another commit

Comment (popomore) [inline src/environment.js]: Saw that :smile:

**Pull request merged**